nixos/modules/services/web-apps/funkwhale.nix

```
@@ -0,0 +1,378 @@
+      '';
+    };
+
+    superUserPasswordFile = mkOption {
+      type = types.nullOr types.path;
+      example = "/run/keys/funkwhale/superuser-password";
+      description = ''
+        File containing the password for the superuser account.
+      '';
+    };
+
+    superUserEmail = mkOption {
+      type = types.str;
+      default = "root@localhost";
+      example = "you@example.com";
+      description = ''
+        Email address for the superuser account.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    assertions = [ {
+      assertion = !(cfg.settings ? "DJANGO_SECRET_KEY");
+      message = ''
+        DJANGO_SECRET_KEY cannot be set through `services.funkwhale.settings`, use `services.funkwhale.secretSettingsFile` instead.
+      '';
+    } ];
+
+    users.groups.funkwhale = {};
+    users.users.funkwhale = {
+      createHome = true;
+      group  = "funkwhale";
+      home = "/var/lib/funkwhale";
+      isSystemUser = true;
+    };
+
+    systemd.targets = {
+      funkwhale = {
+        description = "Funkwhale";
+        wants = [
+          "funkwhale-setup.service"
+          "funkwhale-beat.service"
+          "funkwhale-server.service"
+          "funkwhale-worker.service"
+        ];
+      };
+    };
+
+    systemd.services = let
+      commonUnitConfig = {
+        after = [
+          "network.target"
+          "redis.service"
+          "postgresql.service"
+        ];
+        wantedBy = [
+          "multi-user.target"
+        ];
+        partOf = [
+          "funkwhale.target"
+        ];
+
+        environment = {
+          PYTHONPATH = cfg.package.pythonPath + ":${cfg.package.out}/lib/funkwhale";
+        } // cfg.settings;
+      };
+
+      commonServiceConfig = {
+        User = "funkwhale";
+        Group = "funkwhale";
+        RuntimeDirectory = "funkwhale";
+        StateDirectory = "funkwhale";
+        WorkingDirectory = "/var/lib/funkwhale";
+        EnvironmentFile = [
+          "-/var/lib/funkwhale/secret_key"
+        ] ++ optional (cfg.secretSettingsFile != null) cfg.secretSettingsFile;
+      };
+    in
+    {
+      funkwhale-setup = {
+        description = "Funkwhale setup and migrations";
+        before = [
+          "funkwhale-beat.service"
+          "funkwhale-server.service"
+          "funkwhale-worker.service"
+        ];
+        script = ''
+          if [ -z ''${DJANGO_SECRET_KEY+x} ]; then
+            (
+              umask 0077
+              ${cfg.package.python.interpreter} -c "from django.core.management.utils import get_random_secret_key; print(f'DJANGO_SECRET_KEY=\"{get_random_secret_key()}\"')" > $STATE_DIRECTORY/secret_key
+            )
+            export DJANGO_SECRET_KEY="$(cat $STATE_DIRECTORY/secret_key)"
+          fi
+
+          mkdir -p $STATE_DIRECTORY/bin
+          ln -sf ${manage} $STATE_DIRECTORY/bin/funkwhale
+
+          ${cfg.package}/bin/funkwhale migrate
+          ${cfg.package}/bin/funkwhale collectstatic --no-input
```

```suggestion
          ${cfg.package}/bin/funkwhale migrate
          ${cfg.package}/bin/funkwhale collectstatic --no-input
```
Would it work to add those to path?